schema.ts

```diff
@@ -1293,6 +1293,17 @@ export interface GroupPrincipal {
   groupId: string | number;
 }
 
+/**
+ * This represents a principal that is anyone in that domain
+ *
+ * TODO(sam): Unhide this
+ * @hidden
+ */
+export interface DomainPrincipal {
+  type: PrincipalType.Group;
```

Comment:
What are the implications of reusing Group?  Is/should domain be a new PrincipalType?

Reply:
A group has a list of members which is not how Domain works. Domain is saying if your email address matches this string. To me it is fundamentally different bc of who you are granting access to

Reply:
I think you are answering about moving the domain into GroupPrincipal?  I'm thinking the other way, why isn't there a PrincipalType.Domain?

Reply:
Oh that is an error

Reply:
:-) ok, LGTM now.